lib/media/segment_index.js

```diff
@@ -147,7 +147,9 @@ shaka.media.SegmentIndex.prototype.merge = function(references) {
             i == this.references_.length - 1 &&
             j == references.length - 1,
             'This should be an update of the last segment in a period');
-        newReferences.push(r2);
+        var r = new shaka.media.SegmentReference(r1.position,
+            r2.startTime, r2.endTime, r2.getUris, r2.startByte, r2.endByte);
+        newReferences.push(r);
```

Comment:
I think it could be implemented differently.  If you just increment `i` (not `j`), then the last segment will be treated as new and will be appended with the logic below (I think).

If you do it that way, I would suggest moving the check to the higher else-if chain.